src/link.js

```diff
@@ -1,37 +1,117 @@
 import React, { Component, PropTypes } from 'react';
+
 import { PUSH, REPLACE } from './action-types';
 
-export default class Link extends Component {
-  onClick(event) {
-    event.preventDefault();
-    const { replaceState, dispatch } = this.props;
+const LEFT_MOUSE_BUTTON = 0;
+
+const normalizeLocation = href => {
+  if (typeof href === 'string') {
+    const [pathname, query] = href.split('?');
+    return query ? { pathname, search: `?${query}` } : { pathname };
+  }
+  return href;
+};
+
+const resolveQueryForLocation = ({ linkLocation, persistQuery, currentLocation }) => {
+  const currentQuery = currentLocation &&
+    currentLocation.query;
+
+  // Only use the query from state if it exists
+  // and the href doesn't provide its own query
+  if (
+    persistQuery &&
+    currentQuery &&
+    !linkLocation.search &&
+    !linkLocation.query
+  ) {
+    return { pathname: linkLocation.pathname, query: currentQuery};
+  }
+
+  return linkLocation;
+};
+
+const clickedWithModifier = e =>
+  e.button === LEFT_MOUSE_BUTTON &&
```

Comment:
Latest commit should do the trick! Tested with the enter key: uses `pushState` without modifiers, default if using modifiers 👍  Also checking for the `target` prop